src/query/storage/debugger/storage.go

```diff
@@ -0,0 +1,149 @@
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package debug
+
+import (
+	"context"
+	"errors"
+	"strconv"
+	"time"
+
+	"github.com/m3db/m3/src/query/api/v1/handler/prometheus"
+	"github.com/m3db/m3/src/query/block"
+	"github.com/m3db/m3/src/query/models"
+	"github.com/m3db/m3/src/query/storage"
+	"github.com/m3db/m3/src/query/ts"
+)
+
+type debugStorage struct {
+	seriesList []*ts.Series
+}
+
+// NewStorage creates a new debug storage instance.
+func NewStorage(promReadResp prometheus.PromResp) (storage.Storage, error) {
+	seriesList, err := PromResultToSeriesList(promReadResp, models.NewTagOptions())
+	if err != nil {
+		return nil, err
+	}
+
+	return &debugStorage{
+		seriesList: seriesList,
+	}, nil
+}
+
+func (s *debugStorage) Fetch(
+	ctx context.Context,
+	query *storage.FetchQuery,
+	options *storage.FetchOptions,
+) (*storage.FetchResult, error) {
+	return &storage.FetchResult{
+		SeriesList: s.seriesList,
+	}, nil
+}
+
+func (s *debugStorage) FetchBlocks(
+	ctx context.Context,
+	query *storage.FetchQuery,
+	options *storage.FetchOptions,
+) (block.Result, error) {
+	fetchResult, err := s.Fetch(ctx, query, options)
+	if err != nil {
+		return block.Result{}, err
+	}
+
+	return storage.FetchResultToBlockResult(fetchResult, query)
+}
+
+// PromResultToSeriesList converts a prom result to a series list
+func PromResultToSeriesList(promReadResp prometheus.PromResp, tagOptions models.TagOptions) ([]*ts.Series, error) {
+	results := promReadResp.Data.Result
+	// seriesList := make(map[string]*ts.Series, len(results))
+	seriesList := make([]*ts.Series, len(results))
+
+	for i, result := range results {
+		dps := make(ts.Datapoints, len(result.Values))
+		for i, dp := range result.Values {
+			dps[i].Timestamp = time.Unix(0, int64(dp[0].(float64))*int64(time.Second))
+			s, err := strconv.ParseFloat(dp[1].(string), 64)
+			if err != nil {
+				return nil, err
+			}
+			dps[i].Value = s
+		}
+
+		tagSlice := make([]models.Tag, len(result.Metric))
+		tagsIdx := 0
+		for name, val := range result.Metric {
+			tagSlice[tagsIdx] = models.Tag{
```

Comment:
Can just have `tagSlice = append(tagSlice...` rather than keeping the `tagsIdx` variable

Reply:
Isn't indexing in more optimal?

Reply:
Nevermind, didn't see `models.NewTags()`